extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -319,9 +319,12 @@ export class ProjectsController {
 	 */
 	public async publishToDockerContainer(context: Project | dataworkspace.WorkspaceTreeItem, deployProfile: ILocalDbDeployProfile): Promise<void> {
 		const project: Project = this.getProjectFromContext(context);
+		// Removing the path separator from the image base name to be able to add that in the telemetry. With the separator the name is flagged as user path which is not true
+		// We only need to know the image base parts so it's ok to use a different separator when adding to telemetry
+		const dockerImageNameForTelemetry = deployProfile.localDbSetting?.dockerBaseImage ? deployProfile.localDbSetting.dockerBaseImage.replace(/\//gi, '_') : '';
```

Comment:
fyi you can simplify this a bit more 

`= deployProfile.localDbSetting?.dockerBaseImage?.replace(/\//gi, '_') ?? ''`

Reply:
for my understanding, deployProfile.localDbSetting?.dockerBaseImage? won't return undefined potentially? causing undefined error  for .replace?

Reply:
I'm not sure I understand, if `deployProfile.localDbSetting?.dockerBaseImage` is undefined then it won't call replace at all (it short circuits and will evaluate to undefined, which the `?? ''` then ensures is set as an empty string

Reply:
hmm ok I didn't know that. thanks.

Reply:
Example : https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCm0DyAjAVgRhgXhgbwCgZiYAzAS3gBsATDALnyJJYgFcULqbGAiACWpUQAegDqIAE60AhL2bEAvgWUFQkWAmToATDiYsutBgZYl2nSrUZswNeBTDwaCmMtXrocRFFRoAzPqEhlZ0NnYO5E4uHuAQIFTwAHTCAOYAFFq+6BhJRnQA-EkW+UWS8AAOVACGwPDpAOQiDQA0MA3tAJQwBQXtAKoRjs4yDZ1qcQnJaZk+fjp5oRhFJaFllTV1jc1tHWM9fQ2D9sM0o+OeUykgGVl+-ovcy8UcpUnlVbX1Ta3tXQcDIZREZjIA